### PR TITLE
[Fix] Enable logging of metrics from Callbacks to ConsoleLogging

### PR DESCRIPTION
**Pull request by eracah**
# What does this PR do?

Sample output from examples repo [llm/main.py](https://github.com/mosaicml/examples/blob/main/llm/main.py) run:

```
[batch=1/3]:
	 Train epoch: 0
	 Train trainer/global_step: 0
	 Train trainer/batch_idx: 0
	 Train memory/alloc_requests: 38873
	 Train memory/free_requests: 38760
	 Train memory/allocated_mem: 2666514032128
	 Train memory/active_mem: 15283491328
	 Train memory/inactive_mem: 2903010816
	 Train memory/reserved_mem: 31037849600
	 Train memory/alloc_retries: 5
	 Train trainer/device_train_microbatch_size: 1
	 Train loss/train/total: 12.0570
	 Train metrics/train/LanguageCrossEntropy: 12.0570
	 Train metrics/train/Perplexity: 172309.0625
	 Train throughput/batches_per_sec: 0.0503
	 Train throughput/samples_per_sec: 3.2217
	 Train throughput/device/batches_per_sec: 0.0031
	 Train throughput/device/samples_per_sec: 0.2014
	 Train throughput/tokens_per_sec: 6598.0451
	 Train throughput/device/tokens_per_sec: 412.3778
	 Train throughput/flops_per_sec: 1242426587592322.2500
	 Train throughput/device/flops_per_sec: 77651661724520.1406
	 Train throughput/device/mfu: 0.2489
	 Train wall_clock/train: 19.8653
	 Train wall_clock/val: 0.0000
	 Train wall_clock/total: 19.8653
	 Train lr-DecoupledAdamW/group0: 0.0000
/usr/lib/python3/dist-packages/torch/utils/checkpoint.py:31: UserWarning: None of the inputs have requires_grad=True. Gradients will be None
  warnings.warn("None of the inputs have requires_grad=True. Gradients will be None")
[batch=1/3]:
	 Eval LanguageCrossEntropy: 12.0946
	 Eval Perplexity: 179090.5625
[batch=2/3]:
	 Train epoch: 0
	 Train trainer/global_step: 1
	 Train metrics/eval/LanguageCrossEntropy: 12.0946
	 Train metrics/eval/Perplexity: 179090.5625
	 Train trainer/batch_idx: 1
	 Train memory/alloc_requests: 67172
	 Train memory/free_requests: 66933
	 Train memory/allocated_mem: 5004554130432
	 Train memory/active_mem: 30329803776
	 Train memory/inactive_mem: 1221848064
	 Train memory/reserved_mem: 33632026624
	 Train memory/alloc_retries: 186
	 Train trainer/device_train_microbatch_size: 1
	 Train loss/train/total: 12.0565
	 Train metrics/train/LanguageCrossEntropy: 12.0565
	 Train metrics/train/Perplexity: 172214.4375
	 Train throughput/batches_per_sec: 0.0223
	 Train throughput/samples_per_sec: 1.4281
	 Train throughput/device/batches_per_sec: 0.0014
	 Train throughput/device/samples_per_sec: 0.0893
	 Train throughput/tokens_per_sec: 2924.7470
	 Train throughput/device/tokens_per_sec: 182.7967
	 Train throughput/flops_per_sec: 550736380511142.4375
	 Train throughput/device/flops_per_sec: 34421023781946.4023
	 Train throughput/device/mfu: 0.1103
	 Train wall_clock/train: 64.6801
	 Train wall_clock/val: 15.8078
	 Train wall_clock/total: 80.4879
	 Train lr-DecoupledAdamW/group0: 0.0000
[batch=2/3]:
	 Eval LanguageCrossEntropy: 12.0946
	 Eval Perplexity: 179090.5625
```

# What issue(s) does this change relate to?
fix CO- 1636

### CODE REVIEW

**Comment by mvpatel2000** [comment]
Discussed offline, we're going to remove the trainer arg and that arg to the ConsoleLogger

File: composer/trainer/trainer.py [hunk 1]
```diff
@@ -1130,7 +1130,7 @@ def __init__(
                     'Specifying a `file_path_format_string` to a `RemoteUploaderDownloader` is not currently supported while using `save_latest_filename`. '
                     'Please specify the path formatting via `save_folder`, `save_filename`, and `save_latest_filename`')
 
-        # Callbacks
+        # Callbacks.
```
Comment (mvpatel2000) [inline composer/trainer/trainer.py:1133]:
```suggestion
        # Callbacks
```
idt we use periods anywhere else?

**Review by mvpatel2000 — COMMENTED** [review]
defer to @abhi-mosaic  for approval to verify this solves his issue

File: composer/trainer/trainer.py (hunk 1 above)
Comment (eracah) [reply composer/trainer/trainer.py]: sorry that was just to trigger jenkins again when it was taking too long. forgot to remove it.

**Review by mvpatel2000 — APPROVED** [review]

File: composer/loggers/console_logger.py [hunk 2]
```diff
@@ -84,21 +90,20 @@ def epoch_end(self, state: State, logger: Logger) -> None:
         unit = self.log_interval.unit
 
         if unit == TimeUnit.EPOCH and (cur_epoch % int(self.log_interval) == 0 or cur_epoch == 1):
-            log_dict = {**state.train_metric_values}
```
Comment (dakinggg) [inline composer/loggers/console_logger.py:87]: Why are metrics only cleared on batch/epoch end? Should they be flushed on every event? or at least also `eval_end`?

**Review by dakinggg — APPROVED** [review]
LGTM with one comment

**Comment by eracah** [comment]
> defer to @abhi-mosaic for approval to verify this solves his issue, but lgtm

I sent Abhi a console log output from a mosaicml/examples GPT run and he confirmed over DM that it looks good.

**Pull request merged**